### ISSUE TITLE
More granular inheritance of flags

### ISSUE THREAD

**Issue by GPMueller** [Trick-17/clang-build#66]
As far as I know, currently, when we link a library its include paths are placed in the dependent compile flags and its output directory and its output (archive or shared object) are placed into the dependent link flags.

However (see also https://stackoverflow.com/a/31981843/4069571), there are cases where the dependent target should
- inherit compile or link flags (I don't know an example of this)
- have flags which are not applied to the dependency (e.g. linking shared libraries which the dependency needs, while the dependency is a static library)
- not inherit flags which are applied to the dependency (this is currently our default behaviour)

I believe our default is sensible (corresponding to CMake `PRIVATE` flags.
We could add two subcategories to the `flags` keyword:
```toml
[mylib.flags]
compile = ["-DSOMETHING_PRIVATE"]
[mylib.flags.public]
compile = ["-DSOMETHING_PUBLIC"]
[mylib.flags.interface]
link = ["-lOpenGL"]
```
However, maybe we can think of a better name than `interface`... It's not obvious what that is supposed to mean.

**Comment by GPMueller** [comment]
Implemented with 2d54e0a5ff0dbd391b4112634cd2cecc4e1ab6df on `feature-improved-config`.

**Comment by GPMueller** [comment]
Added `public-include-directories` with ff0cfe4cb293c8b8e4a81781ea34ba84e228827b. Maybe we can think up something better or more intuitive, but it's definitely necessary to distinguish between include directories, which should or should not be inherited by dependent targets.
If it exists, the `include` directory of a target folder is automatically public.
